### PR TITLE
Feature/add mono repo support for MRO

### LINKED ISSUE #187: Implement new interface from ODS-shared lib for supporting monorepos
tracking issue for opendevstack/ods-jenkins-shared-library#242

1. cloning / promoting feature impact (@michaelsauter )
1. doc gen (TIP / R) impact @metmajer)

### LINKED ISSUE #189: MRO should inject / update environment / cluster url? into component DC during environment promotion
When components are promoted we need to inject somehow the environment / the cluster domain? - so components can e.g. put a URL together to the right environment ..  (e.g frontend js calling backend)

@michaelsauter  - any ideas from tailor perspective ... I guess we could have all quickstarters see a DC env param `environment` that one can the overwrite in the `.env` - but that only gets us this far...

### LINKED ISSUE #190: When no jira configured TIx & DTx are marked work in progress
bug in LEvaDocumentChaptersFileService - doc chapters are returned w/o status

### PR DESCRIPTION
**Pull request by clemensutschig**
partially closes #187 
Tested with ML (multi deploy) and also standaline component ..

@metmajer - can you do the TIR doc change? ... you now get builds and deployments as maps :)

```
                        "builds": {
                            "machine-training-service": {
                                "buildId": "machine-training-service-10",
                                "image": "docker-registry.default.svc:5000/odsst-dev/machine-training-service@sha256:0ac5d68e6dc1c4b254a02e877497a3b8199562f89dc04bf3d0cc28fc8efbb051"
                            },
                            "machine-prediction-service": {
                                "buildId": "machine-prediction-service-8",
                                "image": "docker-registry.default.svc:5000/odsst-dev/machine-prediction-service@sha256:b94a6b55ebd5e07c9893df79dd607f7ea0f9f14e9eb41787644f1256a8743b12"
                            }
                        },
                        "deployments": {
                            "machine-training-service": {
                                "podName": "machine-training-service-9-hk7zq",
                                "podNamespace": "odsst-dev",
                                "podCreationTimestamp": "2020-04-15T10:52:12Z",
                                "podEnvironment": "dev",
                                "podNode": ".....",
                                "podIp": ".....",
                                "podStatus": "Running",
                                "deploymentId": "machine-training-service-9",
                                "containers": {
                                    "machine-training-service": "docker-registry.default.svc:5000/odsst-dev/machine-training-service@sha256:0ac5d68e6dc1c4b254a02e877497a3b8199562f89dc04bf3d0cc28fc8efbb051"
                                }
                            },
                            "machine-prediction-service": {
                                "podName": "machine-prediction-service-8-pjwfg",
                                "podNamespace": "odsst-dev",
                                "podCreationTimestamp": "2020-04-15T10:54:39Z",
                                "podEnvironment": "dev",
                                "podNode": ".....",
                                "podIp": ".....",
                                "podStatus": "Running",
                                "deploymentId": "machine-prediction-service-8",
                                "containers": {
                                    "machine-prediction-service": "docker-registry.default.svc:5000/odsst-dev/machine-prediction-service@sha256:b94a6b55ebd5e07c9893df79dd607f7ea0f9f14e9eb41787644f1256a8743b12"
                                }
                            }
                        },
```

### CODE REVIEW

File: src/org/ods/service/OpenShiftService.groovy [hunk 1]
```diff
@@ -300,13 +300,38 @@ class OpenShiftService {
       def pod = [ : ]
         pod.podName = podOCData?.metadata?.name?: "N/A"
         pod.podNamespace = podOCData?.metadata?.namespace?: "N/A"
-        pod.podCreationTimestamp = podOCData?.metadata?.creationTimestamp?: "N/A"
-        pod.podEnvironment = podOCData?.metadata?.labels?.env?: "N/A"
+        pod.podMetaDataCreationTimestamp = podOCData?.metadata?.creationTimestamp?: "N/A"
+        pod.deploymentConfigCreationEnvironment = podOCData?.metadata?.labels?.env?: "N/A"
```
Comment (michaelsauter) [inline src/org/ods/service/OpenShiftService.groovy]: Just as a side note - I realised we do not set this label anymore, and I would suggest to remove reading it here.

**Review by michaelsauter — DISMISSED** [review]

File: src/org/ods/service/OpenShiftService.groovy (hunk 1 above)
Comment (clemensutschig) [reply src/org/ods/service/OpenShiftService.groovy]: removed .env

**Comment by clemensutschig** [comment]
also closes #190

**Comment by clemensutschig** [comment]
closes #189 
closes #190

**Review by metmajer — APPROVED** [review]
Awesome work, thanks @michaelsauter and @clemensutschig!

**Pull request merged**